### PR TITLE
396 test getvisiondatafromnote

### LINKED ISSUE #396: Test GetVisionDataFromNote
- Need to verify distances and rotations work and are robot relative

### PR DESCRIPTION
**Pull request by reslip1**
Closes #396

### CODE REVIEW

**Review by witcpalek — CHANGES_REQUESTED** [review]

File: src/main/deploy/chassis/swervemodule_9997.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<swervemodule xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+  <attributes
+    proportional="5.0"
```
[the remark this answers is not in the document]
Comment (reslip1) [reply src/main/deploy/chassis/swervemodule_9997.xml:4]: this hasn't been tuned yet. It is a placeholder for this bot.

File: src/main/cpp/DragonVision/DragonVision.cpp [hunk 2]
```diff
@@ -420,4 +412,14 @@ bool DragonVision::SetPipeline(DragonCamera::PIPELINE mode, RobotElementNames::C
 DragonCamera::PIPELINE DragonVision::GetPipeline(RobotElementNames::CAMERA_USAGE position)
 {
 	return m_dragonCameraMap[position]->GetPipeline();
+}
+
+void DragonVision::testAndLogVisionData()
+{
+	if (true){
```
[the remark this answers is not in the document]
Comment (reslip1) [reply src/main/cpp/DragonVision/DragonVision.cpp:419]: true. fixed it.

**Review by witcpalek — DISMISSED** [review]

**Comment by witcpalek** [comment]
Please fix build errors

**Comment by reslip1** [comment]
> Please fix build errors

fixed

**Comment by witcpalek** [comment]
Please merge

**Comment by reslip1** [comment]
> Please merge

merged

**Comment by witcpalek** [comment]
Merge is bad; your modifying files that were deleted and deleting newly added files.

Also, should there be changes in code generator files?

**Comment by reslip1** [comment]
> Merge is bad; your modifying files that were deleted and deleting newly added files.
> 
> Also, should there be changes in code generator files?

This was commits to catch up to deletes/changes that were done in main. The files changes will still only be the changes compared to main.

**Comment by witcpalek** [comment]
> > Merge is bad; your modifying files that were deleted and deleting newly added files.
> > Also, should there be changes in code generator files?
> 
> This was commits to catch up to deletes/changes that were done in main. The file changes will still only be the changes compared to main.
There are merge lines in the files and many changes in code generator stuff.  Did you regenerate using code generator?

<<<<<<< HEAD
// This file was automatically generated by the Team 302 code generator version 1.2.3.5
// Generated on Saturday, February 17, 2024 3:03:24 PM
=======
// This file was automatically generated by the Team 302 code generator version 1.3.0.1
// Generated on Sunday, February 18, 2024 10:15:34 PM
>>>>>>> main

#include <string>

#include "PeriodicLooper.h"
#include "utils/logging/Logger.h"
#include "configs/RobotConfigMgr.h"
#include "configs/RobotConfigChassisBot_9997.h"
#include "configs/RobotElementNames.h"
#include "DragonVision/DragonPhotonCam.h"
#include "DragonVision/DragonVision.h"

using std::string;

void RobotConfigChassisBot_9997::DefineMechanisms()
{

}

<<<<<<< HEAD
=======
void RobotConfigChassisBot_9997::DefineLEDs()
{


}

>>>>>>> main

**Comment by reslip1** [comment]
> > > Merge is bad; your modifying files that were deleted and deleting newly added files.
> > > Also, should there be changes in code generator files?
> > 
> > 
> > This was commits to catch up to deletes/changes that were done in main. The file changes will still only be the changes compared to main.
> > There are merge lines in the files and many changes in code generator stuff.  Did you regenerate using code generator?
> 
> # <<<<<<< HEAD
> // This file was automatically generated by the Team 302 code generator version 1.2.3.5
> // Generated on Saturday, February 17, 2024 3:03:24 PM
> 
> // This file was automatically generated by the Team 302 code generator version 1.3.0.1 // Generated on Sunday, February 18, 2024 10:15:34 PM
> 
> > > > > > > > main
> 
> #include
> 
> #include "PeriodicLooper.h" #include "utils/logging/Logger.h" #include "configs/RobotConfigMgr.h" #include "configs/RobotConfigChassisBot_9997.h" #include "configs/RobotElementNames.h" #include "DragonVision/DragonPhotonCam.h" #include "DragonVision/DragonVision.h"
> 
> using std::string;
> 
> void RobotConfigChassisBot_9997::DefineMechanisms() {
> 
> }
> # <<<<<<< HEAD
> 
> void RobotConfigChassisBot_9997::DefineLEDs() {
> 
> }
> 
> > > > > > > > main

thanks! I did see those in the latest build failure. Was still in the process of doing commits for the large merge and those got caught up in the changes. Should be fixed and building now.

**Review by muscatd75 — APPROVED** [review]

**Review by muscatd75 — APPROVED** [review]

**Pull request merged**